### PR TITLE
Fixes to t2dart to make it run within Google

### PR DESCRIPTION
**Pull request by vikerman**
Upstreaming local Google changes to public github repository.
- ts2dart should now handle relative path names as input and basePath. Fixed tests accordingly.
- Removed inlined fs-extra code and switched to using our own mkdirP implementation.
- Also fixed some bugs with mkdirP.

### CODE REVIEW

**Comment by vikerman** [comment]
I changed to use basePath without the cwd. Also added test case to cover all combinations of the input file paths and destination directory being relative and absolute.

Changed mkdirP to the way you suggested.

**Comment by mprobst** [comment]
@rkirov could you review this? I'm stuck in an infinite meeting.

File: lib/mkdirp.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import * as fs from 'fs';
 import * as path from 'path';
 
-
 export default function mkdirP(p: string) {
-  let absPath = process.cwd();
+  p = path.resolve(p);
```
Comment (rkirov) [inline lib/mkdirp.ts]: add a comment, what is this achieving.

**Comment by rkirov** [comment]
looks good, just explain the resolve/relative dance.

**Pull request merged**